bullet_train/docs/upgrades/options.md

```diff
@@ -0,0 +1,93 @@
+* [Upgrade from any version to `1.3.0`](/docs/upgrades/yolo-130.md)
+* [Upgrade from `1.3.x` to `1.4.0`](/docs/upgrades/yolo-140.md)
+
+## About the upgrade process
+
+The vast majority of Bullet Train's functionality is distributed via Ruby gems, but those gems depend on certain
+hooks and initializers being present in the hosting application. Those hooks and initializers are provided by the
+[starter repository](https://github.com/bullet-train-co/bullet_train).
+
+Starting in mid-August of 2023 we began to iterate on how we publish gems and how we ensure that the starter repo
+will get the version of the gems that it expects.
+
+Starting with version `1.3.0` we began to explicitly update the starter repo every time that we released new
+versions of the Bullet Train gems. Unfortunately, at that time we were only making changes to `Gemfile.lock`
+which kind of hides the dependencies, and is often a source of merge conflicts that can be hard to sort out.
+
+Starting with version `1.4.0` we added explicit version numbers for all of the Bullet Train gems to `Gemfile`,
+which is less hidden and is not as prone to having merge conflicts as `Gemfile.lock`.
+
+As a result of these changes, there are a few different ways that you might choose to upgrade your application
+depending on which version you're currently on.
+
+## How to find your current version
+
+1. First open `Gemfile` in your application and look/search for a line that begins with `BULLET_TRAIN_VERSION`.
+   For instance:
+    ```ruby
+    BULLET_TRAIN_VERSION = "1.4.0"
+    ```
+   If your `Gemfile` has such a line, that should be the version you're on. In this case `1.4.0`.
+
+2. If you don't have a `BULLET_TRAIN_VERSION` line in `Gemfile`, then you need to open `Gemfile.lock` and look/search
+   for a line for the `bullet_train` gem. For instance:
+   ```ruby
+   bullet_train (1.2.24)
+   ```
+   In this case the app is on version `1.2.24`
+
+## How to upgrade
+
+Depending on what version you're starting on, and what version you want to get to, you have a few options.
+
+In general your two main options are:
+
+1. Upgrade directly from whatever version you happen to be on all the way to the latest published version.
+2. Do a series of stepwise upgrades from the version you're on to the version you want to get to.
+
+### Upgrade directly from any previous version to the latest version (aka The YOLO Method)
+
+This was the original upgrade method that Bullet Train used for many years. It's still a perfectly useable way of
+upgrading, though it feels a little... let's call it "uncontrolled" to some people. It can definitely lead to some
+hairy merge conflicts if you haven't updated in a long time.
+
+[Read more about The YOLO Method](/docs/upgrades/yolo.md)
+
+### Upgrade from `1.4.0` (or later) to any later version (aka The Standard Stepwise Method)
+
+This is the new standard upgrade method that we recommend. If you've ever upgraded a Rails app from version to version
```

Comment:
This reminds me of https://github.com/Shopify/bootboot, maybe this gem could be useful.